### PR TITLE
Make the Launcher extends FLComponent

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

### Description

Make the Launcher extends FLComponent. Print out the class loading error for module scan.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/app_common/abstract/launcher.py [hunk 1]
```diff
@@ -28,7 +29,7 @@ class LauncherRunStatus:
     NOT_RUNNING = "not_running"
 
 
-class Launcher(ABC):
+class Launcher(FLComponent, ABC):
```
Comment (YuanTingHsieh) [inline nvflare/app_common/abstract/launcher.py:32]: What is the motivation of this change?

Previously I assume users might be writing their own launcher without knowing NVFlare specific concepts.

But maybe that assumption is not correct.

Comment (yhwen) [reply nvflare/app_common/abstract/launcher.py:32]: This will enable the Launcher implementation class be able to handle the NVFlare events. Also be able to create the instance by "name".

Comment (yanchengnv) [reply nvflare/app_common/abstract/launcher.py:32]: We could. On the other hand, I think the SubprocessLauncher should have been a FLComponent any way so that it will benefit from things like event handler.

Anyway, this is no big deal. I think we can call it done after this. Any other classes missing from the table will be added via "path".

Comment (yanchengnv) [reply nvflare/app_common/abstract/launcher.py:32]: About YT's comment, I don't think making Launcher a FLComponent conflicts with the idea that "users might be writing their own launcher without knowing NVFlare specific concepts".  User still will be able to write whatever launcher they want, just now that their launcher can now also benefit from NVFLARE's event handling mechanism.

Comment (yanchengnv) [reply nvflare/app_common/abstract/launcher.py:32]: Besides, YT's statement seems to indicate that knowing NVFLARE specific concepts is a bad thing. I have a serious problem with this! 

Knowing and understanding NVFLARE specific concepts is a wonderful thing, as long as we do not force the user to learn them before doing anything.

**Review by yanchengnv — APPROVED** [review]
LGTM

File: nvflare/app_common/abstract/launcher.py (hunk 1 above)
Comment (YuanTingHsieh) [reply nvflare/app_common/abstract/launcher.py:32]: Sounds good, I agree that as long as we do not force them, then it is good.

Apologize that my phrase was not accurate.

What I mean is we don't want users to be overwhelmed with our complicated (yet great) systems specifics to begin with any customization they might want to do.

This change LGTM.

In that argument, we could make some other components FLComponent as well in later release.
